main.go

```diff
@@ -70,40 +73,20 @@ func main() {
 	}
 	id := strings.TrimSpace(fmt.Sprint(cmd.Stdout)) // docker image id
 
-	// Run protoc (in a container) on each .proto file.
-	//
-	// The explicit PWDs are required to appease protoc's
-	// rather sensitive file name expectations.
-	//
-	// All files in a single protoc invocation must belong
-	// to the same proto package, hence the loop.
-	found := false
-	filepath.Walk("proto", func(path string, info fs.FileInfo, err error) error {
-		if err != nil {
-			return err
-		}
-		if strings.HasSuffix(info.Name(), ".proto") {
-			log.Printf("compiling %s...", path)
-			// We assume pwd does not conflict with some critical part
-			// of the docker image, and volume-mount it.
-			found = true
-			cmd := exec.Command("docker", "run", "-v", pwd+":"+pwd, id,
-				"--proto_path="+pwd+"/proto",
-				"--go_out="+pwd,
-				"--twirp_out="+pwd,
-				"--go_opt=paths=source_relative",
-				pwd+"/"+path,
-			)
-			cmd.Stderr = os.Stderr
-			cmd.Stdout = os.Stderr
-			if err := cmd.Run(); err != nil {
-				log.Fatalf("protoc command (%s) failed: $v", err)
-			}
-		}
-		return nil
-	})
-	if !found {
-		log.Fatal("found no .proto files")
+	// Log the command, neatly.
+	protocArgs := flag.Args()
+	cmdstr := "protoc " + strings.ReplaceAll(strings.Join(protocArgs, " "), pwd, "$(pwd)")
```

Comment:
I'm not loving the idea of computing thet command twice, once here and once on lines 84-85. 

Would this be an appropriate spot to use `cmd.String()` to get the description of what's going to be run?

Reply:
> I'm not loving the idea of computing this command twice, once here and once on lines 84-85.

We're not computing the command twice. This line prints a log message of the protoc command; the lines below construct a docker command that runs protoc with the same arguments. There's nothing redundant.

You could print cmd.String, but you'd have to strip off all the docker prefix, which seems like a more complicated way to achieve the same thing.